libgit/autogit_manager.go

```diff
@@ -51,12 +60,108 @@ func NewAutogitManager(config libkbfs.Config) *AutogitManager {
 		deferLog:               log.CloneWithAddedDepth(1),
 		registeredFBs:          make(map[libkbfs.FolderBranch]bool),
 		repoNodesForWatchedIDs: make(map[libkbfs.NodeID]*repoDirNode),
+		deleteCancels:          make(map[string]context.CancelFunc),
 	}
 }
 
 // Shutdown shuts down this manager.
 func (am *AutogitManager) Shutdown() {
-	// No-op.
+	am.registryLock.Lock()
+	defer am.registryLock.Unlock()
+	am.shutdown = true
+	for _, cancel := range am.deleteCancels {
+		cancel()
+	}
+}
+
+func (am *AutogitManager) removeOldCheckoutsForHandle(
+	ctx context.Context, h *libkbfs.TlfHandle, branch libkbfs.BranchName) {
+	// Make an "unwrapped" FS, so we don't end up recursively entering
+	// the virtual autogit nodes again.
+	fs, err := libfs.NewUnwrappedFS(
+		ctx, am.config, h, branch, "", "", keybase1.MDPriorityNormal)
+	if err != nil {
+		am.log.CDebugf(ctx, "Error making unwrapped FS for TLF %s: %+v",
+			h.GetCanonicalPath(), err)
+		return
+	}
+
+	fi, err := fs.Stat(AutogitRoot)
+	if os.IsNotExist(errors.Cause(err)) {
+		// No autogit repos to remove.
+		return
```

Comment:
It'd be nice to have a log here to make it easy to tell if the cleanup has been finished.

Reply:
I decided not to have a log here, because there was no previous log saying that cleanup had even started, and I didn't want to write to have these log messages that appear on the first autogit access every time.  I could be convinced otherwise though, if you think it's important.

Reply:
I'm fine without a log; don't worry about it!